### PR TITLE
small refactor to make more pythonic

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

**Comment by mgeplf** [comment]
This version produces more entries in the output.json: ex for `DG_GC_type2_AA0091_20181112161509`.  I haven't looked at why.

File: src/assets/config_files/create_singlecellmodeling_structure.py [hunk 1]
```diff
@@ -3,82 +3,104 @@
 Usage: create_singlecellmodeling_structure.py output.json
 '''
 
+import collections
 import json
-import requests
 import logging
 import os
+import requests
```
Comment (antonelepfl) [inline src/assets/config_files/create_singlecellmodeling_structure.py]: Ah using alphabetically order right?

File: src/assets/config_files/create_singlecellmodeling_structure.py [hunk 2]
```diff
@@ -3,82 +3,104 @@
...
+
     return True
 
 
+def filter_required_names(names, required):
+    ret = collections.defaultdict(set)
+    for name in names:
+        for k, ending in required.items():
+            if not name.endswith(ending):
+                continue
```
Comment (antonelepfl) [inline src/assets/config_files/create_singlecellmodeling_structure.py]: One line is not used very much? like `if bla: continue`

File: src/assets/config_files/create_singlecellmodeling_structure.py (hunk 1 above)
Comment (mgeplf) [reply src/assets/config_files/create_singlecellmodeling_structure.py]: yeah, old habits die hard :)

File: src/assets/config_files/create_singlecellmodeling_structure.py [hunk 3]
```diff
@@ -3,82 +3,104 @@
...
             logging.warning('ERROR in cell. Plot not found %s', asset)
 
 
-def _check_models_modification(response):
-    old_list_name = 'old_model_list.txt'
-    if os.path.isfile(old_list_name):
-        x = open(old_list_name).read()
-        if x == response.text:
+def _check_models_modification(new_text):
+    new_text = list(sorted(new_text))
```
[the remark this answers is not in the document]
Comment (mgeplf) [reply src/assets/config_files/create_singlecellmodeling_structure.py]: no, I figure there is not guarantee on what the output from the object storage is, so I sort it proactively myself

File: src/assets/config_files/create_singlecellmodeling_structure.py (hunk 2 above)
Comment (mgeplf) [reply src/assets/config_files/create_singlecellmodeling_structure.py]: yeah, for some reason python style guides usually frown on single line statements, so I have it drilled in my head :)

File: src/assets/config_files/create_singlecellmodeling_structure.py [hunk 4]
```diff
@@ -3,82 +3,104 @@
...
-        _check_consistency(base_url, cell_info[model_name], files_url_list)
     return output_content
 
 
-if __name__ == '__main__':
+def main():
     metadata = create_meta()
     if metadata:
-        logging.debug('Current dir: {0}'.format(os.getcwd()))
+        logging.debug('Current dir: %s', os.getcwd())
```
[the remark this answers is not in the document]
Comment (mgeplf) [reply src/assets/config_files/create_singlecellmodeling_structure.py]: that's a weird one that: basically, logging can lazily compute the the string if it's given the arguments and the format, so it's a small performance thing.  However, that's pretty much the only place that `.format` isn't the preferred way.

**Comment by antonelepfl** [comment]
Ok cool thanks for the learning :)

**Pull request merged**